### PR TITLE
Refactor aztec.js and protocol

### PR DESCRIPTION
**Pull request by PaulRBerg**
### Refactor

- Alphabetise imports and function declarations
- Pull out `eip712` and `secp256k1` and make them independent modules (specifically, `@aztec/typed-data` and `@aztec/secp256k1`)
- Use `keccak256` instead of `sha3`
- Use only one module/function declaration pattern
- Use only one naming scheme for tests, i.e. not `.spec.js`
- Where possible, use arrow functions instead of traditional function definitions

### Breaking

- Rename the `sign` submodule of aztec.js to `signer`

## Types of changes

* Refactor (neither fix or feature)

* Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] Prefix PR title with `[WIP]` if necessary.
-   [x] Add tests to cover changes as needed.
-   [x] Update documentation as needed.

### CODE REVIEW

File: packages/aztec.js/src/proof/joinSplit/helpers.js [hunk 1]
```diff
@@ -55,6 +40,21 @@ helpers.generateCommitmentSet = async ({ kIn, kOut }) => {
     return { commitments, m: inputs.length };
 };
 
+// constructs an AZTEC commitment directly from the setup algorithm's trapdoor key.
+// Used for testing purposes only; we don't know the trapdoor key for the real deal.
+const generateFakeCommitment = async (k, trapdoor) => {
```
Comment (ArnSch) [inline packages/aztec.js/src/proof/joinSplit/helpers.js:45]: Would it make sense to move this to a testing folder/file btw?

Comment (PaulRBerg) [reply packages/aztec.js/src/proof/joinSplit/helpers.js:45]: This is not part of this PR. Will prepare a new one for the `abiEncoder` and `proof` submodules.

File: packages/aztec.js/src/signer/index.js [hunk 2]
```diff
@@ -44,22 +40,21 @@ sign.generateZKAssetDomainParams = function generateZKAssetDomainParams(verifyin
 
 /**
  * Create an EIP712 ECDSA signature over structured data
- * @method signStructuredData
+ * @method signTypedData
```
Comment (ArnSch) [inline packages/aztec.js/src/signer/index.js:43]: Why the change?

Comment (PaulRBerg) [reply packages/aztec.js/src/signer/index.js:43]: That's how it's defined in the official documentation.

**Review by ArnSch — APPROVED** [review]
Seems all good to me, only thing is the structured -> typed change for some of the method names in the signer module, could you explain?
